Compilation stops on building CXX object `FormatTraits.cpp.o`

The compilation process stops with the following output:

```
Scanning dependencies of target FormatTraits
[ 97%] Building CXX object tests/FormatTraits/CMakeFiles/FormatTraits.dir/FormatTraits.cpp.o
/home/luiz/Downloads/Vulkan-Hpp/tests/FormatTraits/FormatTraits.cpp: In function ‘int main(int, char**)’:
/home/luiz/Downloads/Vulkan-Hpp/tests/FormatTraits/FormatTraits.cpp:22:63: error: non-constant condition for static assertion
   22 |  static_assert( vk::blockSize( vk::Format::eR4G4UnormPack8 ) == 1, "Wrong blocksize !" );
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~

/home/luiz/Downloads/Vulkan-Hpp/tests/FormatTraits/FormatTraits.cpp:22:31: error: call to non-‘constexpr’ function ‘uint8_t vk::blockSize(vk::Format)’
   22 |   static_assert( vk::blockSize( vk::Format::eR4G4UnormPack8 ) == 1, "Wrong blocksize !" );
      |                  ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/luiz/Downloads/Vulkan-Hpp/tests/FormatTraits/FormatTraits.cpp:38:34: error: non-constant condition for static assertion
   38 |   static_assert( vk::isCompressed( vk::Format::eBc1RgbSrgbBlock ), "IsCompressed ?" );
      |                  ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/luiz/Downloads/Vulkan-Hpp/tests/FormatTraits/FormatTraits.cpp:38:34: error: call to non-‘constexpr’ function ‘bool vk::isCompressed(vk::Format)’
/home/luiz/Downloads/Vulkan-Hpp/tests/FormatTraits/FormatTraits.cpp:48:54: error: non-constant condition for static assertion
   48 |   static_assert( vk::componentBits( constFormat, 1 ) == 4, "Wrong component bits !" );
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
/home/luiz/Downloads/Vulkan-Hpp/tests/FormatTraits/FormatTraits.cpp:48:35: error: call to non-‘constexpr’ function ‘uint8_t vk::componentBits(vk::Format, uint8_t)’
   48 |   static_assert( vk::componentBits( constFormat, 1 ) == 4, "Wrong component bits !" );
      |                  ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
/home/luiz/Downloads/Vulkan-Hpp/tests/FormatTraits/FormatTraits.cpp:58:60: error: non-constant condition for static assertion
   58 |   static_assert( vk::componentPlaneIndex( constFormat, 8 ) == 0, "Hoo" );
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
/home/luiz/Downloads/Vulkan-Hpp/tests/FormatTraits/FormatTraits.cpp:58:41: error: call to non-‘constexpr’ function ‘uint8_t vk::componentPlaneIndex(vk::Format, uint8_t)’
   58 |   static_assert( vk::componentPlaneIndex( constFormat, 8 ) == 0, "Hoo" );
      |                  ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
make[2]: *** [tests/FormatTraits/CMakeFiles/FormatTraits.dir/build.make:63: tests/FormatTraits/CMakeFiles/FormatTraits.dir/FormatTraits.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:7704: tests/FormatTraits/CMakeFiles/FormatTraits.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

System: Ubuntu 20.04.

--- a/tests/FormatTraits/CMakeLists.txt
+++ b/tests/FormatTraits/CMakeLists.txt
@@ -14,24 +14,22 @@
 
 cmake_minimum_required(VERSION 3.2)
 
-if (NOT TESTS_BUILD_ONLY_DYNAMIC)
-	project(FormatTraits)
+project(FormatTraits)
 
-	set(HEADERS
-	)
+set(HEADERS
+)
 
-	set(SOURCES
-	  FormatTraits.cpp
-	)
+set(SOURCES
+	FormatTraits.cpp
+)
 
-	source_group(headers FILES ${HEADERS})
-	source_group(sources FILES ${SOURCES})
+source_group(headers FILES ${HEADERS})
+source_group(sources FILES ${SOURCES})
 
-	add_executable(FormatTraits
-	  ${HEADERS}
-	  ${SOURCES}
-	)
+add_executable(FormatTraits
+	${HEADERS}
+	${SOURCES}
+)
 
-	set_target_properties(FormatTraits PROPERTIES FOLDER "Tests")
-	target_link_libraries(FormatTraits "${Vulkan_LIBRARIES}")
-endif()
+set_target_properties(FormatTraits PROPERTIES FOLDER "Tests")
+target_link_libraries(FormatTraits "${Vulkan_LIBRARIES}")

--- a/tests/FormatTraits/FormatTraits.cpp
+++ b/tests/FormatTraits/FormatTraits.cpp
@@ -19,7 +19,11 @@
 
 int main( int /*argc*/, char ** /*argv*/ )
 {
+#if VULKAN_HPP_CPP_VERSION < 14
+  assert( vk::blockSize( vk::Format::eR4G4UnormPack8 ) == 1 );
+#else
   static_assert( vk::blockSize( vk::Format::eR4G4UnormPack8 ) == 1, "Wrong blocksize !" );
+#endif
 
   vk::Format format = vk::Format::eR4G4UnormPack8;
   if ( vk::texelsPerBlock( format ) == 1 )
@@ -35,7 +39,11 @@ int main( int /*argc*/, char ** /*argv*/ )
     format = vk::Format::eR4G4UnormPack8;
   }
 
+#if VULKAN_HPP_CPP_VERSION < 14
+  assert( vk::isCompressed( vk::Format::eBc1RgbSrgbBlock ) );
+#else
   static_assert( vk::isCompressed( vk::Format::eBc1RgbSrgbBlock ), "IsCompressed ?" );
+#endif
 
   assert( vk::packed( format ) == 8 );
 
@@ -45,7 +53,11 @@ int main( int /*argc*/, char ** /*argv*/ )
   }
 
   constexpr vk::Format constFormat = vk::Format::eR4G4UnormPack8;
+#if VULKAN_HPP_CPP_VERSION < 14
+  assert( vk::componentBits( constFormat, 1 ) == 4 );
+#else
   static_assert( vk::componentBits( constFormat, 1 ) == 4, "Wrong component bits !" );
+#endif
   assert( vk::componentCount( constFormat ) == 2 );
 
   if ( strcmp( vk::componentName( constFormat, 0 ), "R" ) == 0 )
@@ -55,7 +67,11 @@ int main( int /*argc*/, char ** /*argv*/ )
 
   assert( std::string( vk::componentNumericFormat( constFormat, 1 ) ) == "UNORM" );
 
+#if VULKAN_HPP_CPP_VERSION < 14
+  assert( vk::componentPlaneIndex( constFormat, 8 ) == 0 );
+#else
   static_assert( vk::componentPlaneIndex( constFormat, 8 ) == 0, "Hoo" );
+#endif
 
   if ( vk::planeCount( format ) != 1 )
   {

